src/test/java/pl/debuguj/parkingspacessystem/controllers/ParkingSpaceControllerTest.java

```diff
@@ -0,0 +1,356 @@
+                .append("&stopTime=2017-10-13 23:10:12")
+                .toString();
+        //WHEN
+        mockMvc.perform(post(givenUrl))
+        //THEN
+                .andExpect(status().isBadRequest())
+                .andExpect(content().string(""))
+                .andDo(print())
+                .andReturn();
+    }
+
+    @Test
+    public void shouldReturnNotFoundBecauseOfIncorrectRegistrationNumber() throws Exception {
+        //GIVEN
+        String givenUrl = new StringBuilder()
+                .append(ParkingSpaceController.URI_START_METER)
+                .append("/1111")
+                .append("?driverType=VIP")
+                .append("&startTime=2017-10-13 10:10:12")
+                .append("&stopTime=2017-10-13 23:10:12")
+                .toString();
+        //WHEN
+        mockMvc.perform(post(givenUrl))
+        //THEN
+                .andExpect(status().isNotFound())
+                .andExpect(content().string(""))
+                .andDo(print())
+                .andReturn();
+    }
+
+    @Test
+    public void shouldReturnBadRequestBecauseOfIncorrectEndTime() throws Exception {
+        //GIVEN
+        String givenUrl = new StringBuilder()
+                .append(ParkingSpaceController.URI_START_METER)
+                .append("/11111")
+                .append("?driverType=VIP")
+                .append("&startTime=2017-10-13 23:10:12")
+                .append("&stopTime=2017-10-13 22:10:12")
+                .toString();
+        //WHEN
+        mockMvc.perform(post(givenUrl))
+        //THEN
+                .andExpect(status().isBadRequest())
+                .andExpect(content().string(""))
+                .andDo(print())
+                .andReturn();
+    }
+
+    @Test
+    public void shouldReturnThatVehicleIsRegisteredInSystem() throws Exception {
+        //GIVEN
+        String givenUrlSetData = new StringBuilder()
+                .append(ParkingSpaceController.URI_START_METER)
+                .append("/22222")
+                .append("?driverType=VIP")
+                .append("&startTime=2017-10-15 10:10:12")
+                .append("&stopTime=2017-10-15 12:15:12")
+                .toString();
+
+        mockMvc.perform(post(givenUrlSetData));
+
+        String givenUrl = new StringBuilder()
+                .append(ParkingSpaceController.URI_CHECK_VEHICLE)
+                .append("/22222")
+                .append("?currentDate=2017-10-15 10:28:48")
+                .toString();
+        //WHEN
+        mockMvc.perform(get(givenUrl))
+        //THEN
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8_VALUE))
+                .andExpect(content().string("true"))
+                .andDo(print())
+                .andReturn();
+    }
+
+    @Test
+    public void shouldReturnThatVehicleIsNotRegisteredInSystem() throws Exception {
+        String givenUrl = new StringBuilder()
```

Comment:
Zamiast ręcznego budowania Stringów, `String.format()` mógłby być wygodniejszy w użyciu

Reply:
Z mojej wiedzy wynikało, że StringBuilder jest najwydajniejszy, więc z przyzwyczajenia zawsze go stosuje. O wydajności String.format() doczytam, bo wygodniejszego wygląda.

Reply:
Owszem jest, ale od jakiegoś czasu nie jest w żaden sposób lepszy niż zwykła konkatenacja Stringów za pomocą operator `+` - 

Wyciąg z sekcji 15.18.1 Java Language Specification:
`An implementation may choose to perform conversion and concatenation in one step to avoid creating and then discarding an intermediate String object. To increase the performance of repeated string concatenation, a Java compiler may use the StringBuffer class or a similar technique to reduce the number of intermediate String objects that are created by evaluation of an expression.`

Poza tym, w aplikacjach, gdzie odbywa się serializacja/deserializacja i przesyłanie komunikatów przez sieć, narzut wydajnościowy z okazji operacji na Stringach jest statystycznie pomijalny

Reply:
Cenne info, dzięki.